### DISCUSSION TITLE
BPF_TRACE_MAX_PROGS | BPFTRACE_MAX_BPF_PROGS: Limit enable by app level or another limit is used at kernel level

### DISCUSSION THREAD

**Discussion by kevin-intri-form3** [bpftrace/bpftrace#2823] [Q&A]
Recently we found this macro [`#define BPF_TRACE_MAX_PROGS 64`](https://elixir.bootlin.com/linux/v5.15.10/source/kernel/trace/bpf_trace.c#L1723) and we don't know if that overlap and limit this [BPFTRACE_MAX_BPF_PROGS](https://github.com/iovisor/bpftrace/blob/096235a10ebe8c955375a4755d1a42619135f767/docs/reference_guide.md#98-bpftrace_max_bpf_progs).

Do you know if they are correlated or the kernel one has the actual limit?

Thanks.

**Comment by danobi** [comment]
Hi, the two are unrelated. The naming is just a coincidence. It looks like the kernel BPF_TRACE_MAX_PROGS controls how many progs can be attached to a single perf event. While the bpftrace BPFTRACE_MAX_PROGS controls how many programs bpftrace will codegen.

**Reply by kevin-intri-form3** [reply]
Thanks @danobi!
